lib/Dialect/FIRRTL/Transforms/LowerClasses.cpp

```diff
@@ -945,28 +992,23 @@ void LowerClassesPass::lowerClass(om::ClassOp classOp, FModuleLike moduleLike,
     if (!needsClone && !propertyOperands && !propertyResults)
       continue;
 
-    // Actually clone the op over to the OM Class.
-    builder.clone(op, mapping);
+    if (isa<PropAssignOp>(op) &&
+        dyn_cast<BlockArgument>(cast<PropAssignOp>(op).getDest())) {
```

Comment:
nit: if you're just doing a boolean check, I think you could do `isa<BlockArgument>(...)`. `dyn_cast` is also useful for the first part of this predicate...

```
if (auto propAssign = dyn_cast<PropAssignOp>(op) &&
    isa<BlockArgument>(propAssign.getDest()))
    ...
```

I can't find the reference, but I think something in the [LLVM coding standards](https://llvm.org/docs/CodingStandards.html) discourages using the `propAssign` within the same if condition, so maybe this would be better:

```
if (auto propAssign = dyn_cast<PropAssignOp>(op))
    if (isa<BlockArgument>(propAssign.getDest()))
        ...
```